### PR TITLE
bump hpa to v2 and adjust non required fields

### LINKED ISSUE #1566: Handle HPAs in preferred version (v2beta2 -> v2)
**Description**

We should use preferred version of the resources all the time. In case of HPAs, that would be `v2`, while we are using `v2beta2`. Smooth upgrade is impossible, since in `v2` the new one a field (`metrics`) was removed / made optional. Let's research if an upgrade is viable.

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- bump version of hpa to `v2`
- remove metrics and minPods from `template` for create and from tests,
- remove handler for unsupported version.

**Related issue(s)**
Closes #1566

### CODE REVIEW

**Comment by mrCherry97** [comment]
Everything should work fine, going threw [kubernetes examples of HPA usage](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/) everything is fine. Busola is displaying everything properly, etc.

File: src/resources/HorizontalPodAutoscalers/HorizontalPodAutoscalerDetails.js [hunk 1]
```diff
@@ -99,14 +101,104 @@ export function HorizontalPodAutoscalerDetails(props) {
...
+            <LayoutPanel.Body>
+              <LayoutPanelRow
+                name={t('hpas.headers.stabilizationWindowSeconds')}
+                value={
+                  scaleUp?.stabilizationWindowSeconds ?? EMPTY_TEXT_PLACEHOLDER
+                }
+              />
+              <LayoutPanelRow
+                name={t('hpas.headers.selectPolicy')}
+                value={scaleUp?.selectPolicy ?? EMPTY_TEXT_PLACEHOLDER}
```
Comment (Lyczeq) [inline src/resources/HorizontalPodAutoscalers/HorizontalPodAutoscalerDetails.js:145]: If you check whether `scaleUp` exists, there's no need to use optional chaining in '140 and '145. Error won't be thrown.

File: src/resources/HorizontalPodAutoscalers/HorizontalPodAutoscalerDetails.js [hunk 2]
```diff
@@ -99,14 +101,104 @@ export function HorizontalPodAutoscalerDetails(props) {
...
+              <LayoutPanelRow
+                name={t('hpas.headers.stabilizationWindowSeconds')}
+                value={
+                  scaleDown?.stabilizationWindowSeconds ??
+                  EMPTY_TEXT_PLACEHOLDER
+                }
+              />
+              <LayoutPanelRow
+                name={t('hpas.headers.selectPolicy')}
+                value={scaleDown?.selectPolicy ?? EMPTY_TEXT_PLACEHOLDER}
```
Comment (Lyczeq) [inline src/resources/HorizontalPodAutoscalers/HorizontalPodAutoscalerDetails.js:179]: If you check whether `scaleDown` exists, there's no need to use optional chaining in '173 and '179. Error won't be thrown.

**Review by Lyczeq — CHANGES_REQUESTED** [review]

**Review by Lyczeq — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**